### ISSUE TITLE
ClassSerializerInterceptor is not working

### ISSUE THREAD

**Issue by Maples7** [nestjs/nest#1589]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

@UseInterceptors(ClassSerializerInterceptor) is not working

## Expected behavior

@UseInterceptors(ClassSerializerInterceptor) should respect @Exclude and other Decorators in class-transformer

## Minimal reproduction of the problem with instructions

users.entity.ts:
```typescript
@Entity('user')
export class UserEntity {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({
    type: 'varchar',
    length: 100,
  })
  @MinLength(6)
  @MaxLength(100)
  @Exclude({ toPlainOnly: true })
  password: string;

  constructor(partial: Partial<UserEntity>) {
    Object.assign(this, partial);
  }
}
```

users.controller.ts:
```typescript
@Controller('users')
@UseInterceptors(ClassSerializerInterceptor)
export class UsersController {
  constructor(
    private readonly usersService: UsersService,
  ) {}

  @Post('register')
  async create(@Body() createUserDto: CreateUserDto): Promise<UserEntity> {
    const { password } = createUserDto;
    const newUser = await this.usersService.createUserByHimself(
      password,
    );
    return newUser;
  }
}
```

users.service.ts:
```typescript
@Injectable()
export class UsersService {
  private readonly logger = new Logger('UsersService');

  async createUserByHimself(
    password: string,
  ): Promise<UserEntity> {
    const newUser = new UserEntity({
      password,
    });
    try {
      const savedUser = await getManager().transaction(
        async transactionalEntityManager => {
          return await transactionalEntityManager.save(newUser);
        },
      );
      return savedUser;
    } catch (err) {
      this.logger.error('Fail to create a new user', err);
    }
  }
}
```

main.ts:
```typescript
const app = await NestFactory.create(AppModule, new FastifyAdapter());
// ...
app.useGlobalPipes(
  new ValidationPipe({
    whitelist: true,
    transform: true,
    disableErrorMessages: isProduction,
  }),
);
```

## What is the motivation / use case for changing the behavior?

Response Object serialization.

## Environment
<pre><code>
Nest version: 5.4.0
 
For Tooling issues:
- Node version: v10.15.0 
- Platform:  Mac
</code></pre>

I am not sure if it's a bug or anywhere in my code is wrong. Help please!

**Comment by Maples7** [comment]
By the way, I have my own `transform.interceptor.ts`:
```typescript```
import { ExecutionContext, Injectable, NestInterceptor } from '@nestjs/common';
import { Observable } from 'rxjs';
import { map } from 'rxjs/operators';

@Injectable()
export class TransformInterceptor implements NestInterceptor {
  intercept(
    context: ExecutionContext,
    call$: Observable<any>,
  ): Observable<any> {
    return call$.pipe(
      map(data => ({
        success: true,
        data,
      })),
    );
  }
}
```

And in my `main.ts`:
```typescript
app.useGlobalInterceptors(new TransformInterceptor());
```

I am pretty sure it's the conflict between my own TransformInterceptor and ClassSerializerInterceptor, so my question has become to how to avoid it.

**Comment by Maples7** [comment]
That's true! It's the conflict between my own TransformInterceptor and ClassSerializerInterceptor, then I think this issue #1464 may related with this one.

**Comment by adesege** [comment]
I also have this issue. In my case, I do not have a custom interceptor and I have toggled between global and local interceptors but none worked.

**Comment by kamilmysliwiec** [comment]
Duplicate of #1464, caused by #1016. It will be fixed in the incoming 6.0.0 release. Sorry for the inconvenience.
